### ISSUE TITLE
update decodestring to decodebytes in ssh2john

### ISSUE THREAD

**Issue by maniac-en** [openwall/john#4557]
### IMPORTANT ###
This is not a support forum, it's a bug tracker. If you don't understand the difference, please do NOT open an issue. For questions and support, review recent postings on the john-users mailing list at https://www.openwall.com/lists/john-users/ and then subscribe to the list at https://www.openwall.com/john/#lists and post your message to the list by e-mailing `john-users at lists.openwall.com`.

We are not interested in bugs in your distro-supplied and/or several years old Jumbo. Please do not open an issue unless you tried the latest version from HERE first.

### Steps to reproduce ###
Try to be clear about your environment and what you are doing. If possible, share a sample hash or file that can be used to reproduce.

![image](https://user-images.githubusercontent.com/45049766/105758624-75f90b00-5f75-11eb-85f0-75f00cc6ec06.png)

### System configuration ###
> uname -r: 5.10.9-arch1-1
> Installed JTR via arch community repo.

Attach details about your OS and about JtR, including:
- `$ ./john --list=build-info`.
```
Version: 1.9.0-jumbo-1
Build: linux-gnu 64-bit x86_64 AVX AC MPI + OMP
SIMD: AVX, interleaving: MD4:3 MD5:3 SHA1:1 SHA256:1 SHA512:1
System-wide exec: /usr/lib/john
System-wide home: /usr/share/john
Private home: ~/.john
CPU tests: AVX
CPU fallback binary: john-non-avx
$JOHN is /usr/share/john/
Format interface version: 14
Max. number of reported tunable costs: 4
Rec file version: REC4
Charset file version: CHR3
CHARSET_MIN: 1 (0x01)
CHARSET_MAX: 255 (0xff)
CHARSET_LENGTH: 24
SALT_HASH_SIZE: 1048576
SINGLE_IDX_MAX: 2147483648
SINGLE_BUF_MAX: 4294967295
Effective limit: Number of salts vs. SingleMaxBufferSize
Max. Markov mode level: 400
Max. Markov mode password length: 30
gcc version: 10.2.0
GNU libc version: 2.32 (loaded: 2.32)
OpenCL headers version: 2.2
Crypto library: OpenSSL
OpenSSL library version: 01010108f	(loaded: 01010109f)
OpenSSL 1.1.1h  22 Sep 2020	(loaded: OpenSSL 1.1.1i  8 Dec 2020)
GMP library version: 6.2.0	(loaded: 6.2.1)
File locking: fcntl()
fseek(): fseek
ftell(): ftell
fopen(): fopen
memmem(): System's
```
- `$ ./john --list=opencl-devices` (if applicable).

**Comment by solardiz** [comment]
Thanks for reporting, but quoting the issue template, which you also quoted (and didn't read?) -

```
We are not interested in bugs in your distro-supplied and/or several years old Jumbo. Please do not open an issue unless you tried the latest version from HERE first.
```

This specific issue was fixed with 660de879bd06a2edbe228b7c742d4a98ec989248 in November 2019, and then there were some more commits to that script.

I take this as a reminder for us to make a new numbered release in not too distant future.

@anthraxx might also consider updating the Arch package to bleeding-jumbo once in a while, although I understand this is too much to expect from a distro, and it's our "job" to make timely releases.

**Comment by anthraxx** [comment]
@solardiz  thanks for pinging me, unfortunately the general rule of thumb for distro packaging is that we indeed only package releases. bumping every now and then is a huge anti pattern during distro packaging. We may however backport individual patches if they are reasonably easy to backport. All in all it would be much easier to just get frequent upstream releases, especially since all the valuable work would hit more end users :cat:
